### PR TITLE
Feature/split tiles row

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: src/client/components/split-tile/add-split.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        label: dimension.title,
+        value: dimension
+      };
+    });
+
+  return <AddTile<Dimension>
+    containerStage={menuStage}
+    onSelect={appendSplit}
+    tiles={tiles} />;
+
```
Comment (bigsergey) [inline src/client/components/split-tile/add-split.tsx]: unnecessary empty line

File: src/client/components/split-tile/split-tiles-row.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  drop = (e: React.DragEvent<HTMLElement>) => {
+    if (!this.canDrop()) return;
+    e.preventDefault();
+    this.setState({ dragPosition: null });
+
+    const split = this.draggingSplit();
+    if (!split) return;
+
+    const { clicker, essence: { splits } } = this.props;
+    let dragPosition = this.calculateDragPosition(e);
```
Comment (erykpiast) [inline src/client/components/split-tile/split-tiles-row.tsx]: why not ternary expression?

File: src/client/components/split-tile/split-tiles-row.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    if (dragPosition.replace === this.maxItems()) {
+      dragPosition = new DragPosition({ insert: dragPosition.replace });
+    }
+    if (dragPosition.isReplace()) {
+      clicker.changeSplits(splits.replaceByIndex(dragPosition.replace, split), VisStrategy.FairGame);
+    } else {
+      clicker.changeSplits(splits.insertByIndex(dragPosition.insert, split), VisStrategy.FairGame);
```
Comment (bigsergey) [inline src/client/components/split-tile/split-tiles-row.tsx]:
```
const currentDragPosition = dragPosition.isReplace() ? dragPosition.replace : dragPosition.insert;
clicker.changeSplits(splits.insertByIndex(currentDragPosition), VisStrategy.FairGame);
```

File: src/client/components/split-tile/split-tiles.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      split={split}
+      dimension={dimension}
+      removeSplit={removeSplit}
+      updateSplit={updateSplit}
+      open={split.equals(openedSplit)}
+      openMenu={openMenu}
+      closeMenu={closeMenu}
+      dragStart={dragStart}
+      containerStage={menuStage}
+      essence={essence} />;
```
Comment (bigsergey) [inline src/client/components/split-tile/split-tiles.tsx]: maybe `...props`?

File: src/client/components/split-tile/split-tiles-row.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    e.preventDefault();
+    this.setState({ dragPosition: null });
+
+    const split = this.draggingSplit();
+    if (!split) return;
+
+    const { clicker, essence: { splits } } = this.props;
+    let dragPosition = this.calculateDragPosition(e);
+    if (dragPosition.replace === this.maxItems()) {
+      dragPosition = new DragPosition({ insert: dragPosition.replace });
```
Comment (adrianmroz) [inline src/client/components/split-tile/split-tiles-row.tsx]: early return with `changeSplits`

File: src/client/components/split-tile/split-tiles-row.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    if (dragPosition.isReplace()) {
+      clicker.changeSplits(splits.replaceByIndex(dragPosition.replace, split), VisStrategy.FairGame);
```
Comment (adrianmroz) [inline src/client/components/split-tile/split-tiles-row.tsx]: extract as `replace` and `insert` methods

File: src/client/components/split-tile/split-tiles.tsx (hunk 4 above)
Comment (adrianmroz) [reply src/client/components/split-tile/split-tiles.tsx]: Not really, some props shouldn't be passed so you would need to cut them out, creating new object and still adding few more. I think that way it's more readable.

File: src/client/components/split-tile/split-tiles-row.tsx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+export class SplitTilesRow extends React.Component<SplitTilesRowProps, SplitTilesRowState> {
+  private items = React.createRef<HTMLDivElement>();
+
+  state: SplitTilesRowState = {};
+
+  private maxItems(): number {
+    const { menuStage, essence: { splits: { splits } } } = this.props;
+    return menuStage && getMaxItems(menuStage.width, splits.count());
```
Comment (erykpiast) [inline src/client/components/split-tile/split-tiles-row.tsx]: Is `menuStage` a number? `false && 1` returns `false`, that clearly isn't a number. The output type of this function looks like `Stage | number` to me.

Comment (adrianmroz) [reply src/client/components/split-tile/split-tiles-row.tsx]: more like `falsy | number`

Comment (erykpiast) [reply src/client/components/split-tile/split-tiles-row.tsx]: In `SplitTilesRowProps`, `menuStage` isn't optional, so I though `Stage` may be falsy somehow. So, maybe `SplitTilesRowProps.menuStage` should be declared with `?`?

Comment (adrianmroz) [reply src/client/components/split-tile/split-tiles-row.tsx]: It should.

**Pull request merged**